samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -153,31 +166,46 @@ public ApplicationStatus status(StreamApplication streamApp) {
   }
 
   /**
-   * Create the Coordination needed by the application runner.
-   * @return {@link CoordinationUtils}
-   * @throws Exception exception
+   * Block until the application finishes
+   */
+  public void waitForFinish() {
+    try {
+      shutdownLatch.await();
+    } catch (Exception e) {
+      log.error("Wait is interrupted by exception", e);
+      throw new SamzaException(e);
+    }
+  }
+
+  /**
+   * Create the {@link CoordinationUtils} needed by the application runner.
+   * @return an instance of {@link CoordinationUtils}
```

Comment:
nit: update return annotation to include the possibility of returning "null" value